### PR TITLE
Failing tests should stop build when publishing

### PR DESCRIPTION
**Pull request by davegreen**
If psake is called with the publish task and one or more tests fail, an exception will be generated to abort the build and prevent the publish from occurring. As per #123

### CODE REVIEW

**Comment by daviwil** [comment]
You can ignore the build failure, I'm still in the process of getting it working :)

**Comment by daviwil** [comment]
Now that I'm trying to get the AppVeyor build going, I can't see any reason why the `Test` task shouldn't fail when any Pester test fails.  Based on the changes in this PR, If I invoke psake with only the `Test` task selected, psake will still report that the build succeeded even if Pester tests have failed.

I'd recommend taking out the `Publish` task check and just make the `Test` task fail if any tests failed.  Also change the task dependency for `Install` to `Build`.  What do you think, @rkeithhill?

File: build.psake.ps1 [hunk 1]
```diff
@@ -243,7 +243,11 @@ Task Test -depends Build {
     Import-Module Pester
     try {
         Microsoft.PowerShell.Management\Push-Location -LiteralPath $TestRootDir
-        Invoke-Pester
+        $TestResult = Invoke-Pester -PassThru -Verbose:$VerbosePreference
```
Comment (rkeithhill) [inline build.psake.ps1]: Have you tried this from VSCode?  When there are Pester test failures, we have a VSCode problem matcher that deciphers the Pester output and populates the VSCode Problems panel.  This looks like VSCode will no longer see the output.  It's all captured in $TestResult.

**Comment by rkeithhill** [comment]
@daviwil yup, agree with both points.

File: build.psake.ps1 (hunk 1 above)
Comment (daviwil) [inline build.psake.ps1]: Pester still writes out the test output even when -PassThru is being used.  I just tried it when messing with the AppVeyor stuff, works fine.

Comment (rkeithhill) [inline build.psake.ps1]: Output stream output is captured in the variable.  Are you saying Invoke-Pester also writes directly to the host?

Comment (rkeithhill) [inline build.psake.ps1]: And indeed it does write to the host.  OK, no problem then.

**Comment by davegreen** [comment]
OK, sorted :)

**Pull request merged**